clang/lib/CAS/CASOptions.cpp

```diff
@@ -75,3 +75,15 @@ CASOptions::getOrCreateCASAndHideConfig(DiagnosticsEngine &Diags) {
 
   return CAS;
 }
+
+void CASOptions::ensurePersistentCAS() {
+  assert(!IsFrozen);
+  switch (getKind()) {
+  case UnknownCAS:
```

Comment:
Should this be `llvm_unreachable` since you have the assert above?
```
llvm_unreachable("Cannot ensure persistent CAS if it's unknown / frozen");
```

Reply:
Done👍